### PR TITLE
feat: Update Kotlin SDK to use Gson instead of Jackson

### LINKED ISSUE #825: Android App should Gracefully Handle Extra SDK Parameters
Currently the Android app does not handle changes to new parameters added to the SDK. We need to handle any new parameters gracefully.

Analyzing the code it seems that the deserialization of json is handled by the SDK (Transport.kt:customClient(...)).

Possible fix would be to add disable(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES) to JacksonSerializer while setting up the serializer.
```
install(JsonFeature) {
    serializer = JacksonSerializer() {
        disable(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES)
    }
}
```

### PR DESCRIPTION
**Pull request by mistryrakesh**
This PR updates the kotlin SDK to use Gson instead of Jackson for deserialization (We were already using gson to serialize).

This fixes:
- Jackson serialization bug [Jackson renames primitive boolean field by removing 'is'](https://stackoverflow.com/questions/32270422/jackson-renames-primitive-boolean-field-by-removing-is) resulting in much clean and concise code.
- In some cases prevents breaking of the SDK due to API changes by ignoring unknown fields ([How to decode JSON with unknown field using Gson?](https://stackoverflow.com/a/20442943)). e.g. When SDK is generated from an older instance but has to access a newer instance.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/looker-open-source/sdk-codegen/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Appropriate docs were updated (if necessary)

Fixes #825 🦕

### CODE REVIEW

**Review by jkaster — CHANGES_REQUESTED** [review]
Great update. Nice to have cleaner and more flexible code.

```sh
bin/smoke kotlin
```
passes.

Can you add a test that verifies the graceful handling of unknown fields?

**Comment by mistryrakesh** [comment]
Hey @jkaster thanks for reviewing!

I have added a smoke test (`SmokeTest.testIgnoreUnknownFields()`) which basically fetches all users and deserializes it into `DummyUser` object. The `DummyUser` class is a subset of the `User` class. When the server sends a response `User` object the Gson deserializer ignores extra properties and creates the `DummyUser` object.

Using the Jackson deserializer the test fails with error:
```
java.lang.Error: GET /users com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "avatar_url" (class DummyUser), not marked as ignorable (4 known properties: "display_name", "id", "credentials_api3", "email"])
 at [Source: (StringReader); line: 1, column: 1260] (through reference chain: java.lang.Object[][0]->DummyUser["avatar_url"]).message
java.lang.AssertionError: java.lang.Error: GET /users com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "avatar_url" (class DummyUser), not marked as ignorable (4 known properties: "display_name", "id", "credentials_api3", "email"])
 at [Source: (StringReader); line: 1, column: 1260] (through reference chain: java.lang.Object[][0]->DummyUser["avatar_url"]).message
	at org.junit.Assert.fail(Assert.java:88)
	at kotlin.test.junit.JUnitAsserter.fail(JUnitSupport.kt:56)
	at kotlin.test.Asserter$DefaultImpls.assertTrue(Assertions.kt:226)
	at kotlin.test.junit.JUnitAsserter.assertTrue(JUnitSupport.kt:30)
	at kotlin.test.Asserter$DefaultImpls.assertTrue(Assertions.kt:236)
	at kotlin.test.junit.JUnitAsserter.assertTrue(JUnitSupport.kt:30)
	at kotlin.test.AssertionsKt__AssertionsKt.assertTrue(Assertions.kt:40)
	at kotlin.test.AssertionsKt.assertTrue(Unknown Source)
	at TestSmoke.testIgnoreUnknownFields(TestSmoke.kt:212)
```

**Comment by jkaster** [comment]
@mistryrakesh This is great stuff! I really appreciate you contributions here.

File: kotlin/src/test/TestSmoke.kt [hunk 1]
```diff
@@ -186,4 +188,39 @@ class TestSmoke {
...
+            received = true
+        } catch (e: java.lang.Error) {
+            assertTrue(false, "$e.message")
+        }
+        assertEquals(true, received)
+    }
 }
+
+/**
+ * Dummy user data class based on [User] but with a subset of its properties.
```
Comment (jkaster) [inline kotlin/src/test/TestSmoke.kt:219]: Perfect

**Review by jkaster — APPROVED** [review]
Nicely done! It's great to have better resilience for the Kotlin SDK.

File: kotlin/src/test/TestSmoke.kt (hunk 1 above)
Comment (mistryrakesh) [reply kotlin/src/test/TestSmoke.kt:219]: Thanks!

**Pull request merged**